app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -0,0 +1,112 @@
+class ManageIQ::Providers::Vmware::CloudManager::RefreshParser < ManageIQ::Providers::CloudManager::RefreshParser
+  def self.ems_inv_to_hashes(ems, inv, options = nil)
+    new(ems, inv, options).ems_inv_to_hashes
+  end
+
+  def initialize(ems, inv, options = Config::Options.new)
+    @ems                 = ems
+    @inv                 = inv
+    @data                = {}
+    @data_index          = {}
+    @options             = options
+  end
+
+  def process_collection(collection, key)
+    @data[key] ||= []
+
+    collection.each do |item|
+      uid, new_result = yield(item)
+      next if uid.nil?
+
+      @data[key] << new_result
+      @data_index.store_path(key, uid, new_result)
+    end
+  end
+
+  def ems_inv_to_hashes
+    log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{@ems.name}] id: [#{@ems.id}]"
+
+    $log.info("#{log_header}...")
+
+    get_vms
+
+    $log.info("#{log_header}...Complete")
+
+    @data
+  end
+
+  private
+
+  def get_vms
+    process_collection(@inv[:vms], :vms) { |vm| parse_vm(vm) }
+  end
+
+  def parse_vm(vm)
+    status = vm.status
+
+    uid           = vm.id
+    name          = vm.name
+    guest_os      = vm.operating_system
+    bitness       = !(vm.operating_system =~ /64-bit/).nil? ? 64 : 32
```

Comment:
Think this can be simplified to just `vm.operating_system =~ /64-bit/ ? 64 : 32` without the `! nil?`